generator/transit_generator.hpp

```diff
@@ -1,9 +1,90 @@
+#pragma once
+
+#include "geometry/point2d.hpp"
+
+#include "base/macros.hpp"
+
+#include "3party/jansson/myjansson.hpp"
+
+#include <cstdint>
 #include <string>
+#include <type_traits>
+#include <vector>
 
 namespace routing
 {
 namespace transit
 {
-void BuildTransit(std::string const & mwmPath, std::string const & srtmDir);
+class DeserializerFromJson
+{
+public:
+  DeserializerFromJson(json_struct_t * node) : m_node(node) {}
+
+  template<typename T>
+  typename std::enable_if<std::is_integral<T>::value || std::is_enum<T>::value>::type
+      operator()(T & t, char const * name = nullptr)
+  {
+    GetField(t, name);
+  }
+
+  template<typename T>
+  typename std::enable_if<std::is_floating_point<T>::value>::type operator()(T & t, char const * name = nullptr)
+  {
+    NOTIMPLEMENTED();
+  }
+
+  void operator()(std::string & s, char const * name = nullptr) { GetField(s, name); }
+  void operator()(m2::PointD & p, char const * name = nullptr);
+
+  template <typename T>
+  void operator()(std::vector<T> & vs, char const * name = nullptr)
+  {
+    auto * arr = my::GetJSONOptionalField(m_node, name);
+    if (arr == nullptr)
+      return;
+
+    if (!json_is_array(arr))
+      MYTHROW(my::Json::Exception, ("The field", name, "must contain a json array."));
+    size_t const sz = json_array_size(arr);
+    vs.resize(sz);
+    for (size_t i = 0; i < sz; ++i)
+    {
+      DeserializerFromJson arrayItem(json_array_get(arr, i));
+      arrayItem(vs[i]);
+    }
+  }
+
+  template<typename T>
+  typename std::enable_if<std::is_class<T>::value>::type operator()(T & t, char const * name = nullptr)
+  {
+    t.Visit(*this);
+  }
+
+private:
+  template <typename T>
+  void GetField(T & t, char const * name = nullptr)
+  {
+    if (name == nullptr)
+    {
+      // |name| is not set in case of array items
+      FromJSON(m_node, t);
+    }
+    else
+    {
+      json_struct_t * field = my::GetJSONOptionalField(m_node, name);
+      if (field == nullptr)
+        return; // No field |name| at |m_node|.
+      FromJSON(field, t);
+    }
+  }
+
+  json_struct_t * m_node;
+};
+
+/// \brief Builds transit section at mwm.
+/// \param mwmPath relative or full path to built mwm. The name of mwm without extension is considered
```

Comment:
s/built/an/
If you need to mention specific sections of the mwm that you need, please list them explicitly.

Reply:
> s/built/an/

Done

> If you need to mention specific sections of the mwm that you need, please list them explicitly.

Are you talking about mwm sections which I need to build transit section?

Reply:
Yes. You wrote "built mwm" which I read as "a fully-built mwm", while in fact you probably can call this code earlier.

Reply:
Ok. Now I'm just keeping in mwm information which comes from json. So it's ok to built the section on early stages now. Later I'll need: 
- mapping from osm id to feature id (The mapping is creating after our features is created);
- and information about boarders (to cut transit graph by mwms);

Do you want me to write the comment right now or later when I use the information for building transit section?